### PR TITLE
[FT] Support for Arduino Framework

### PR DESCRIPTION
**Pull request by dhillondeep**
Now you can do:
```
default: main
   create:
     main:
       src: main
       platform: avr
       framework: arduino
```

* It includes a hotfix for executing native programs that use stdin
* Added `Standard` field that can be used to set C/C++ standard
* Removed using a standard as a framework for `native` platform

Merge #57 PR first and then this builds upon that one
**Update:** Merged #57, Overwriting #57

### CODE REVIEW

**Review by Mogball — CHANGES_REQUESTED** [review]

File: cmd/wio/commands/run/cmake/cmake.go [hunk 1]
```diff
@@ -83,7 +142,7 @@ func GenerateNativeCmakeLists(
         "CPP_STANDARD":               cppStandard,
         "C_STANDARD":                 cStandard,
         "TARGET_NAME":                (*target).GetName(),
-        "FRAMEWORK":                  framework,
+        "FRAMEWORK":                  "CPP",
```
[the remark this answers is not in the document]
Comment (dhillondeep) [reply cmd/wio/commands/run/cmake/cmake.go]: true made changes

File: toolchain/arduino/revisions.txt [hunk 2]
```diff
@@ -0,0 +1,1860 @@
+ARDUINO 1.8.5 2017.09.29
+
+(bugfix service release)
+
+[ide]
+* Added workaround for menu visualization bug in MacOSX 10.13 beta. Thanks @puybaret
+* Fixed bug for negative-font-size.
+* New/Rename tabs now allows names starting with a number.
+
+ARDUINO 1.8.4 2017.08.23
```
[the remark this answers is not in the document]
Comment (Mogball) [reply toolchain/arduino/revisions.txt:10]: why not just make it empty

File: cmd/wio/types/general-types.go [hunk 3]
```diff
@@ -93,6 +94,7 @@ type AppTarget struct {
     Src         string
     Platform    string               `yaml:"platform,omitempty"`
     Framework   string               `yaml:"framework,omitempty"`
+    Standard    string               `yaml:"standard,omitempty"`
```
[the remark this answers is not in the document]
Comment (Mogball) [reply cmd/wio/types/general-types.go:97]: I've never specified c and C++ standard in one line..

**Comment by Mogball** [comment]
meh I'm okay with the cpp standard stuff

if you can add a test package/app that shows it working too

**Review by Mogball — APPROVED** [review]

File: toolchain/arduino/revisions.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (dhillondeep) [reply toolchain/arduino/revisions.txt:10]: it needs a version num from there like apparently it uses Arduino IDE version or something. I did not wanna mess with it.

File: cmd/wio/types/general-types.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dhillondeep) [reply cmd/wio/types/general-types.go:97]: Well for native we may add support for some other language like golang lmao so a single Standard keeps it consistent

**Pull request merged**